plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -2032,11 +2032,20 @@ int getReservedCpuMHZ(VirtualMachineTO vmSpec) {
 
     // Pair<internal CS name, vCenter display name>
     private Pair<String, String> composeVmNames(VirtualMachineTO vmSpec) {
-        String vmInternalCSName = vmSpec.getName();
-        String vmNameOnVcenter = vmSpec.getName();
-        if (_instanceNameFlag && vmSpec.getHostName() != null) {
-            vmNameOnVcenter = vmSpec.getHostName();
+
+        String vmInternalCSName = null;
+        String vmNameOnVcenter  = null;
+        if(vmSpec != null)
```

Comment:
@rafaelweingartner Currently. it is not possible. Because user does not generate it. Cloud stack generates it only if it has all required info. otherwise it does not call this function.